src/bin/version/main.rs

```diff
@@ -0,0 +1,101 @@
+extern crate core;
+
+use colored::Colorize;
+use std::ffi::OsString;
+use std::path::Path;
+use std::{env, fs};
+
+const SDKMAN_DIR_ENV_VAR: &str = "SDKMAN_DIR";
+const DEFAULT_SDKMAN_HOME: &str = ".sdkman";
+const VERSION_FILE: &str = "var/version";
+
+fn main() {
+    let sdkman_dir = infer_sdkman_dir();
+    let version = locate_version_file(sdkman_dir).and_then(read_content);
+    match version {
+        Some(content) => println!("\n{} {}", "SDKMAN".yellow(), content.yellow()),
+        None => std::process::exit(exitcode::CONFIG),
+    }
+}
+
+fn infer_sdkman_dir() -> String {
+    match env::var(SDKMAN_DIR_ENV_VAR) {
+        Ok(s) => s,
+        Err(_) => fallback_sdkman_dir(),
+    }
+}
+
+fn fallback_sdkman_dir() -> String {
+    dirs::home_dir()
+        .map(|dir| dir.join(DEFAULT_SDKMAN_HOME))
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_string()
+}
+
+fn locate_version_file(base_dir: String) -> Option<OsString> {
+    Some(
+        Path::new(base_dir.as_str())
+            .join(VERSION_FILE)
+            .into_os_string(),
```

Comment:
Is the conversion actually needed? Most fs-related APIs accept a "Path-like" object.

Reply:
Unfortunately yes. You can't get from a `PathBuf` to a normal string, as it could be different on each OS.